### DISCUSSION TITLE
feat: add fish speech support

### DISCUSSION THREAD

**Discussion by codeMonkey-shin** [open-webui/open-webui#24847] [Ideas]
https://github.com/fishaudio/fish-speech

The local tts app has an API server, and it would be nice to have the ability to integrate with that server.

In particular, the advantage is that you can set the voice you want to communicate through example voices, and that very fast inferences are possible.

**Comment by tjbck** [comment]
Looks great, will investigate shortly!

**Comment by a1558878** [comment]
> Looks great, will investigate shortly!

What's the result?

**Comment by codgician** [comment]
Wondering what is the latest status here as I see #11230 has been closed. Any further work or discussion needed? Thanks.

**Comment by ayutaz** [comment]
For those looking for a lightweight, self-hosted multilingual TTS option that works with Open WebUI's existing OpenAI-compatible TTS integration:

[piper-plus](https://github.com/ayutaz/piper-plus) provides an OpenAI-compatible `/v1/audio/speech` endpoint out of the box (via Docker or FastAPI), supporting 6 languages (JA/EN/ZH/ES/FR/PT) in a single ~75MB ONNX model — no GPU required.

Quick setup with Docker:
```
docker compose -f docker/python-inference/docker-compose.yml up
```

Then point Open WebUI's TTS settings to `http://<host>:8000/v1/audio/speech`.

Key features:
- Runs on CPU (Raspberry Pi-capable)
- 6-language single model, no eSpeak-ng (MIT license)
- Voice cloning from reference audio
- ~75MB FP16 ONNX model

Demo: https://huggingface.co/spaces/ayousanz/piper-plus-demo
Models: https://huggingface.co/ayousanz/piper-plus-base
